### PR TITLE
build: replace travis with github actions

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+name: Test and publish
+
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+    tags: [ v.* ]
```
Comment (raboof) [inline .github/workflows/validate.yml:7]: Are we sure this works? Something similar didn't work for Akka gRPC which triggered the change at https://github.com/akka/akka-grpc/commit/bba0d14c539c0f9835d6937ae06e0473c92ebfab , but I don't know whether that was about the list syntax or the tag regex

Comment (jrudolph) [reply .github/workflows/validate.yml:7]: No not yet :)

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+jobs:
+  # share matrix between workflows, see https://github.community/t/how-to-share-matrix-between-jobs/128595/8
```
Comment (jrudolph) [inline .github/workflows/validate.yml:10]: Matrix sharing is probably not needed as publishing doesn't require it right now. Removing again...

File: .github/workflows/validate.yml (hunk 1 above)
Comment (jrudolph) [reply .github/workflows/validate.yml]: It's not a regex but a glob pattern so it should be `v*`.

**Comment by jrudolph** [comment]
I gave up trying to extract scala versions from the build (sbt outputs on stderr and I didn't manage to capture stderr inside of github actions for some reason).

File: .github/workflows/validate.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        with:
+          java-version: adopt@1.8
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v5
+
+      - name: Publish
+        run: sbt ci-release
+        env:
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
```
Comment (raboof) [inline .github/workflows/validate.yml:99]: doesn't this also need `PGP_PASSPHRASE`? or is that optional when the secret is not password-protected?

**Review by raboof — APPROVED** [review]

File: .github/workflows/validate.yml (hunk 3 above)
Comment (jrudolph) [reply .github/workflows/validate.yml:99]: Should be optional. I used it elsewhere without passphrase.

Comment (octonato) [reply .github/workflows/validate.yml:99]: I can confirm it's optional. We have a few projects without it and it works.

**Comment by jrudolph** [comment]
Publishing worked for test release 0.18-RC1:

https://github.com/lightbend/genjavadoc/runs/3076948355

:fireworks:

**Pull request merged**